Sources/SwiftKuery/ConnectionPool.swift

```diff
@@ -23,8 +23,23 @@ import Dispatch
 
 // MARK: ConnectionPool
 
-/// A connection pool implementation.
-/// The pool is FIFO.
+/**
+ This class implements a first in first out connection pool. The pool maintains a cache of `ConnectionPoolConnection` instances, which can be reused for future requests, to enhance performance.
+ ### Usage Example: ###
+This example shows how to create a ConnectionPool manually. Two closures to generate and release Connection instances are defined. A connection pool is then initialized with the defined closures and options. A single Connection is then retrieved from the pool.
```

Comment:
I don't see Enrique's comment addressed here? His comment was - "I feel this is quite confusing...We should explain that the Plugins for PostgreSQL and MySQL implement this class and you can create a ConnectionPool directly by calling the specific plugin." "And that if you want to do it manually you can do like this"...